safehttp/response.go

```diff
@@ -49,3 +51,24 @@ type TemplateResponse struct {
 // NoContentResponse is sent to the commit phase when it's initiated from
 // ResponseWriter.NoContent.
 type NoContentResponse struct{}
+
+// NewJSONResponse creates a JSON response from a data object. This should be a
+// valid JSON object that can be serialised and written to the
+// http.ResponseWriter using a JSON encoder.
+func NewJSONResponse(data interface{}) JSONResponse {
+	return JSONResponse{Data: data}
+}
+
+// NewTemplateResponse creates a TemplateResponse from a Template, its data and
+// the name to function mappings. These will be pre-processed and then written
+// to the http.ResponseWriter, if deemed safe. If the funcMap is non-nil, its
+// elements will override the existing name to function mappings in the
+// template. An attempt to define a new name to function mapping that is not
+// already in the template will result in a panic.
+func NewTemplateResponse(t Template, data interface{}, funcMap map[string]interface{}) TemplateResponse {
+	return TemplateResponse{
+		Template: &t,
+		Data:     &data,
```

Comment:
Why are you taking a reference here?

Reply:
Right, that is not actually needed. When we designed the commit phase we've assumed the token and nonce injection is going to be done directly by the XSRF and CSP plugin which is not actually the case.